sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AdaptingCircularBufferCounter.java

```diff
@@ -73,13 +73,13 @@ public boolean increment(int index, long delta) {
 
     if (index > endIndex) {
       // Move end, check max size
-      if (index - startIndex + 1 > backing.length()) {
+      if ((long) index - (long) startIndex + 1L > backing.length()) {
```

Comment:
When recording `Double.MIN_VALUE` and `Double.MAX_VALUE` at scale 20 in the same histogram, the two indexes here differ so much that it causes an integer overflow. As a result, the if statement evaluates to false when it should be true. This causes an error later on because the histogram didn't downscale, and it tries to create an array of negative length. 

Casting to `long` fixes this, as it can't possibly overflow at the max scale of 20.

Reply:
Think it makes sense to reduce some noise with just one cast

`(long) index - startIndex + 1`

Reply:
Good suggestion, all done.

Reply:
Might be worth an explanatory comment.